v-next/hardhat-ethers/src/internal/signers/deep-copy.ts

```diff
@@ -0,0 +1,54 @@
+import { HardhatError } from "@ignored/hardhat-vnext-errors";
+import { deepClone, isObject } from "@ignored/hardhat-vnext-utils/lang";
+
+const PRIMITIVE = [
+  "bigint",
+  "boolean",
+  "function",
+  "number",
+  "string",
+  "symbol",
+];
+
+export async function deepCopy<T = any>(value: T): Promise<T> {
+  // The function 'deepClone' from 'hardhat-utils' cannot be used to replace this function, it won't properly clone
+  // the value.
```

Comment:
I'd add an explanation about the "why" to this comment

Reply:
I saw that dis code comes from v2, so keeping it makes sense

Reply:
I added a comment to better explain why `deepClone` cannot be used to replace `deepCopy`: https://github.com/NomicFoundation/hardhat/pull/5787/commits/8348448bf64bebb099fb158c6e9bb268211167b7